subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/DefaultTaskContainerTest.groovy

```diff
@@ -751,10 +751,10 @@ class DefaultTaskContainerTest extends Specification {
         ex.message == "Failing constructor"
 
         and:
-        !provider.isPresent()
+        provider.isPresent()
```

Comment:
❌ I'm not sure if once the task has failed, it should be considered not present. If an eager task fails creation, you won't be able to "find" it in the container. I feel the behavior should be similar with lazy task. The difference is you should be able to "find" the provider but it should say there's nothing present. Realizing the provider again will cause a null task to be returned.